docs/content/how-to-tune-workers.md

```diff
@@ -0,0 +1,112 @@
+3. `sticky_cache_size` and `workflow_active_thread_count` report the size of the Workflow cache and the number of cached Workflow threads.
+
+_Note: To have access to the metrics mentioned above, JavaSDK ≥ 1.8.0 is required._
+
+## Configuration
+
+The following options are defined on `WorkerOptions` and are applicable for each Worker separately:
+
+1. `maxConcurrentWorkflowTaskExecutionSize` and `maxConcurrentActivityExecutionSize` define the number of total available slots for that Worker.
+2. `maxConcurrentWorkflowTaskPollers` (_JavaSDK: `workflowPollThreadCount`_)and `maxConcurrentActivityTaskPollers` (_JavaSDK: `activityPollThreadCount`_) define the number of pollers performing poll requests waiting on Workflow / Activity task queue and delivering the tasks to the executors.
+
+The Workflow Cache is created and shared between all the workers. It’s designed to limit the amount of resources used by the cache for the whole host/process. So the options are defined on `WorkerFactoryOptions` in JavaSDK and in `worker` package in GoSDK:
+
+1. `WorkerFactoryOptions#workflowCacheSize` (_GoSDK: `worker.setStickyWorkflowCacheSize`_)defines the maximum number of cached Workflows Executions. Each cached Workflow contains at least one Workflow thread and its resources (memory, etc).
+2. `maxWorkflowThreadCount` defines the maximum number of Workflow threads.
+
+These options limit the resource consumption of the in-memory Workflow cache. Workflow cache options are shared between all Workers, because the Workflow cache is something that has to do with the resource consumption of the whole host, like memory and the total amount of threads, and should be limited per JVM.
+
+## Tuning
+
+The steps are arranged in the recommended order of execution.
+
+### Provision enough hosts/resources
+
+If currently provisioned Worker hosts are fully utilized (near full CPU usage, high load average, etc), additional Workers hosts have to be provisioned to increase the capacity of the Workers pool.
+
+### Worker slots and executor sizing
+
+The main area of tuning should be the number of Workflow slots. If:
+
+1. the Worker hosts are underutilized (there are no bottlenecks on CPU, load average, etc), and
+2. the `worker_task_slots_available` metric from the corresponding Worker type often shows a depleted number of available Worker slots,
+
+then consider increasing the maximum number of working slots by adjusting `maxConcurrentWorkflowTaskExecutionSize` or `maxConcurrentActivityExecutionSize`.
+
+### Poller count
+
+_Note: Adjustments to pollers are rarely needed and rarely make a difference. Please consider this step only after adjusting Worker slots in the previous step. The only scenario in which the pollers’ adjustment makes sense is when there is a significant network latency between the Workers and Temporal Server._
```

Comment:
Maybe move this section to be after the others then?

Reply:
I added clarifications in the update, I think that the current order makes the most sense. Every next step makes sense only after checking each of the previous ones.